### PR TITLE
Create README.md

### PR DESCRIPTION
**Pull request by michaelliu2**
Add README with checklist

### CODE REVIEW

**Comment by Hornswoggles** [comment]
+1

File: README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+This is a line_only_client that writes to a TCP socket with the following specs (unchecked means TODO):
+
+- [x] writes messages delimited by '\r\n' and waits for response
+- [x] uses a connection pool
+- [x] connection retry
+- [x] times out waiting for response and closes connection
+- [ ] logs timeout and message request failure exceptions
+- [ ] handles OK/fail response
+- [ ] retries after timeout with exponential backoff
```
Comment (spheromak) [inline README.md:11]: would reword this to on error.
